stitches/dev_matchup.py

```diff
@@ -8,155 +8,283 @@
-        d = historical_data.copy()
-        d["experiment"] = scn
-        frames.append(d)
-
-    frames.append(future_data)
-    data = pd.concat(frames)
-
-    # Index the data frame by the year, so that the rolling mean respects the years
-    data.index = data['year']
-
-    # Now calculate the rolling mean that uses a centered window.
-    group_by = ['model', 'experiment', 'ensemble', 'variable']
-    rslt = data.groupby(group_by)['value'].rolling(size, center=True).mean().reset_index()
-
-    return rslt
+  # Relabel the historical values so that there is a continuous rolling mean between the
+  # historical and future values.
+  # #######################################################
+  # Subset the historical data
+  historical_data = input_data[input_data["experiment"] == "historical"].copy()
+
+  # Create a subset of the future data
+  future_data = input_data[input_data["experiment"] != "historical"].copy()
+  future_scns = set(future_data["experiment"].unique())
+
+  frames = []
+  for scn in future_scns:
+    d = historical_data.copy()
+    d["experiment"] = scn
+    frames.append(d)
+
+  frames.append(future_data)
+  data = pd.concat(frames)
+
+  # Index the data frame by the year, so that the rolling mean respects the years
+  data.index = data['year']
+
+  # Now calculate the rolling mean that uses a centered window.
+  group_by = ['model', 'experiment', 'ensemble', 'variable']
+  rslt = data.groupby(group_by)['value'].rolling(size, center=True).mean().reset_index()
+
+  return rslt
+
+
+
+def chunk_ts(df, n):
+
+  """ Format a data frame into an array of data frames containing data for n-sized years of successive data.
+
+  :param df:     data frame of climate data to chunk into different periods
+  :type df:     pandas DataFrame
+
+  :param n:    the size of the windows to chunk into separate periods
+  :type n:    int
+
+  :return:    pandas DataFrame identical to df with the addition of a chunk column
+
+  # TODO hmmm do we want this function to also work on ESM data? If so do we need to change this function??
+  # TODO does it need to be for only one variable?
+  # TODO do we like chunks or are these windows?
+  """
+
+  # Check the inputs
+  req_cols = {'year', 'variable', 'value'}
+  col_names = set(df.columns)
+  if not (req_cols.issubset(col_names)):
+    raise TypeError(f'Missing columns from "{df}".')
+  if not (len(df["variable"].unique()) == 1):
+    raise TypeError(f'Multiple variables discovered in "{df}"')
+
+  # TODO How do we want to handle when the length of the time series cannot be split up into even chunks?
+  # Add a column of data that categorizes the data into the different periods, right now we allow for
+  # a list that is smaller than n at the end.
+  # Save a copy of the length of the list
+  yr_len = len(df["year"].unique())
+
+  # Determine the number of unique n sized chunks, right now we assume that at most a chunk
+  # can have n entries or less.
+  n_chunks = math.ceil(yr_len / n)
```

Comment:
@abigailsnyder would it make sense to include a check that says that the window has to be an odd number?

Reply:
I think at some point, probably. Or maybe a check so it throws a message on even number years being like 'this is all centered rolling averages so things get worse/weird when you have even numbers'?